packages/flutter/lib/src/material/popup_menu.dart

```diff
@@ -964,6 +966,19 @@ class _PopupMenuRoute<T> extends PopupRoute<T> {
     return super.createAnimation();
   }
 
+  @override
+  TickerFuture didPush() {
+    final TickerFuture future = super.didPush();
+    // Registers a callback that allows programmatic dismissal of the menu,
+    // This forces the menu to close with a null result
+    registerCloseCallback?.call(() {
+      if (isActive) {
+        navigator?.pop();
+      }
+    });
```

Comment:
What's the benefit of passing this callback up? Since showMenu needs a context, you can just use that to to pop it if necessary. If the showMenu Future is expected to resolve a value of type <T> then this also might give the suggestion that the callback returned from registerCloseCallback would handle that, when it doesn't.